packages/core/test/expo-plugin/modifyAppDelegate.test.ts

```diff
@@ -0,0 +1,177 @@
+import type { ExpoConfig } from '@expo/config-types';
+
+import { warnOnce } from '../../plugin/src/utils';
+import { modifyAppDelegate } from '../../plugin/src/withSentryIOS';
+
+// Mock dependencies
+jest.mock('@expo/config-plugins', () => ({
+  ...jest.requireActual('@expo/config-plugins'),
+  withAppDelegate: jest.fn((config, callback) => callback(config)),
+}));
+
+jest.mock('../../plugin/src/utils', () => ({
+  warnOnce: jest.fn(),
+}));
+
+interface MockedExpoConfig extends ExpoConfig {
+  modResults: {
+    path: string;
+    contents: string;
+    language: 'swift' | 'objc';
+  };
+}
+
+const objcContents = `#import "AppDelegate.h"
+
+@implementation AppDelegate
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
+{
+  self.moduleName = @"main";
+
+  // You can add your custom initial props in the dictionary below.
+  // They will be passed down to the ViewController used by React Native.
+  self.initialProps = @{};
+
+  return [super application:application didFinishLaunchingWithOptions:launchOptions];
+}
+
+@end
+`;
+
+const objcExpected = `#import "AppDelegate.h"
+#import <RNSentry/RNSentry.h>
+
+@implementation AppDelegate
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
+{
+  [RNSentrySDK start];
+  self.moduleName = @"main";
+
+  // You can add your custom initial props in the dictionary below.
+  // They will be passed down to the ViewController used by React Native.
+  self.initialProps = @{};
+
+  return [super application:application didFinishLaunchingWithOptions:launchOptions];
+}
+
+@end
+`;
+
+const swiftContents = `import React
+import React_RCTAppDelegate
+import ReactAppDependencyProvider
+import UIKit
+
+@main
+class AppDelegate: RCTAppDelegate {
+  override func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil) -> Bool {
+    self.moduleName = "sentry-react-native-sample"
+    self.dependencyProvider = RCTAppDependencyProvider()
+    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
+  }
+}`;
+
+const swiftExpected = `import React
+import React_RCTAppDelegate
+import ReactAppDependencyProvider
+import UIKit
+import RNSentry
+
+@main
+class AppDelegate: RCTAppDelegate {
+  override func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil) -> Bool {
+    RNSentrySDK.start()
+    self.moduleName = "sentry-react-native-sample"
+    self.dependencyProvider = RCTAppDependencyProvider()
+    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
+  }
+}`;
+
+describe('modifyAppDelegate', () => {
+  let config: MockedExpoConfig;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    // Reset to a mocked Swift config after each test
+    config = {
```

Comment:
Let's deep copy or use func to create the config to avoid comparing the same reference int he test with no modification.

Reply:
Good catch 👍 Updated in 744993c8255ca160e41673f01cc09f7e58eb3c69